### PR TITLE
Add test block chain json file

### PR DESCRIPTION
**Pull request by SWvheerden**
## Description
Added test json file for use in block chain testing
add partial eq to merkle mountain range

## Motivation and Context
This gives us a test block chain pre generated for testing proposes with transactions and spending keys for those. The transaction builder is slow enough that you cannot create blocks with transactions in meaningful numbers in unit tests. The file provided has 50 blocks with at least 4 transactions per block. 

## How Has This Been Tested?
Unit tests comparing the serialised/deserialised blocks

## Types of changes
* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [x] New Tests
* [ ] Documentation

## Checklist:
* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.

### CODE REVIEW

File: base_layer/core/src/block.rs [hunk 1]
```diff
@@ -30,17 +30,18 @@ use crate::{
...
     types::*,
 };
+use serde::{Deserialize, Serialize};
 use tari_crypto::{
     commitment::HomomorphicCommitmentFactory,
     keys::{PublicKey, SecretKey},
     range_proof::RangeProofService,
 };
-use tari_utilities::byte_array::ByteArray;
+use tari_utilities::byte_array::*;
```
Comment (CjS77) [inline base_layer/core/src/block.rs]: Generally I prefer not using wildcards, so it's easy to read exactly where imported objects comes from

File: base_layer/core/src/types.rs [hunk 2]
```diff
@@ -111,3 +118,47 @@ impl ByteArray for RangeProof {
         &self.0
     }
 }
+
+impl fmt::Display for RangeProof {
```
Comment (CjS77) [inline base_layer/core/src/types.rs]: You may as well move RangeProof to its own file, since these impls clearly don't go in types.rs

File: base_layer/core/tests/chain/chain.json [hunk 3]
```diff
@@ -0,0 +1 @@
...7e6225fbcfd25a354194af51c428c62","proof":"b88cdeaebe1f1b1781b938298bce7a829e8ba061d0d7b3129859b1798fd1be0d2e67d91c326544c43c7efee8ca664d5465b7bcc0f6ca6aacfc0a68540977740c9a82a3a4f9e68ea99161f356f0dc225d71ca0ded833924163687ea82fe8a1e7c76428268a5f172ef5628f128f1424d1cb2019fb7f2fea876db3e206b69a8c5020d7166716b0f7cdbde732d1a975e0996a6a48bad4479cb633e4b56793074840a7fb2b6d24c02dfbfa8b4a00e2848cce693455ac9a0489677919092f98f086f0837c24f28d89c9369d7ccb1673b32bd0ee51b490154e9cbcd7dde6e6abd1806020473b1e74d3c835eb31c41e1a3a82116eddac471b3f3cbb24ce119152a89824f8e3f3902b2ec0818abbf62b3e9d85137a05cd3eeb448b8514335b3c90132ee2bce4a55f372609b568ac50ef6dd5aff7ea26949cdbe84b244c962d96b23d2736b60a2d95aaf6e1af42c1d4206743c728cd3296ab7caa13c8cf643666ca88d680dd6b684725e2e8ee4faf7166d797e5d04266089e6d1468cd9eeeb667166fb8549a0dc4433b467e5f808f8e3d0a0e7663d8d2a52a41d2a121ce853dfdf0e84cb125e5ac50c30ea272e9af515c25384c26addfd879d58178df6bbf3876588c1235baaeb748b58a4a47722fc0105b7810949796ff755edb782800669cbc92a2c2745a03d0d98600ee9e12054793aacf11f17931e70664b6b6fb77420f091650c6e5a7e72b9660f688c7f92d0c7fae02d9f521c3ee1172b212fe397087ffe04aec019ac8b710c568a7660339d222e870df430ce7fe26327eb95aa5d081eba0074e11012dd5b
```
Comment (CjS77) [inline base_layer/core/tests/chain/chain.json]: This is one line. Can you pretty-print this?

File: infrastructure/merklemountainrange/src/treemap_ser.rs [hunk 4]
```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 The Tari Project
```
Comment (CjS77) [inline infrastructure/merklemountainrange/src/treemap_ser.rs]: Technically, this could go in a separate PR

Comment (SWvheerden) [reply infrastructure/merklemountainrange/src/treemap_ser.rs]: I could technically, but this was added here because it required it.

File: base_layer/core/tests/chain/chain.json (hunk 3 above)
Comment (SWvheerden) [reply base_layer/core/tests/chain/chain.json]: If we have to pretty print it, it will make deserializing difficult. Currently its just serialize -> deserialize. 
Many viewers will pretty it when viewing. Thats just the raw file.
